eahub/profiles/models.py

```diff
@@ -482,220 +296,58 @@ def geocode(self):
 
-    def has_cause_area_details(self):
-        cause_area_details_exist = [
-            len(self.cause_areas) > 0,
-            len(self.cause_areas_other) > 0,
-            len(self.giving_pledges) > 0,
-            self.available_to_volunteer,
-        ]
-        return any(cause_area_details_exist)
+    def get_tags_speech_topic_formatted(self) -> List[str]:
+        return [tag.name for tag in self.tags_speech_topic.all()]
 
-    def has_career_details(self):
-        career_details_exist = [
-            len(self.expertise_areas),
-            len(self.expertise_areas_other),
-            self.open_to_job_offers,
-        ]
-        return any(career_details_exist)
-
-    def has_community_details(self):
-        community_details_exist = [
-            len(self.organisational_affiliations) > 0,
-            self.local_groups.exists(),
-            self.user.localgroup_set.exists(),
-            self.available_as_speaker,
-            len(self.topics_i_speak_about) > 0,
-            self.offering,
-            self.looking_for,
-        ]
-        return any(community_details_exist)
+    def get_local_groups_formatted(self) -> List[str]:
+        return [group.name for group in self.local_groups.all()]
 
-    def get_is_organiser(self):
-        return self.user.localgroup_set.exists()
+    def get_organizer_of_local_groups_formatted(self) -> List[str]:
+        return [group.name for group in self.user.localgroup_set.all()]
 
-    def get_can_receive_message(self):
-        return self.is_approved and self.is_public and self.allow_messaging
-
-    def convert_to_row(self, field_names):
-        values = []
-        for field in field_names:
-            if "_other" in field:
-                continue
-            elif field == "cause_areas":
-                values.append(self.get_pretty_cause_areas())
-            elif field == "expertise_areas":
-                values.append(self.get_pretty_expertise())
-            elif field == "career_interest_areas":
-                values.append(self.get_pretty_career_interest_areas())
-            elif field == "organisational_affiliations":
-                values.append(self.get_pretty_organisational_affiliations())
-            elif field == "giving_pledges":
-                values.append(self.get_pretty_giving_pledges())
-            elif field == "local_groups":
-                values.append(self.get_pretty_local_groups())
-            else:
-                values.append(getattr(self, field))
-        return values
-
-    def _format_enum_array_for_searching(
-        self, enum_values_list: List[Union[enum.Enum, str, int]], enum_cls: enum.Enum
-    ) -> List[str]:
-        enum_labels: List[str] = []
-        for enum_value_raw in enum_values_list:
-            enum_value = int(enum_value_raw)
-            enum_labels.append(enum_cls.values[enum_value].label)
-        return enum_labels
-
-    @staticmethod
-    def get_exportable_field_names():
```

Comment:
Export on profiles misses a number of fields, among others:
- Organisational affiliation
- speech topic
- open to volunteer
- social media pages

The custom code you removed was relying on this method here to get all those fields, and the 3rd party library we're using now seems to be missing them.

Reply:
I know, we don't need to handle those fields in a custom manner anymore, as they're handled automatically when added to the `fields = []` attribute of the resource class.

I added few more fields that you mentioned, if you want more you can add them as well.

Does that sound sufficient?

Reply:
Fine for now, when we want to export data to other parties we probably want all fields, but we can change that then